packages/jest-cli/src/SearchSource.js

```diff
@@ -33,7 +33,7 @@ type Options = {|
   withAncestor?: boolean,
 |};
 
-export type PathPattern = {|
+export type PathPatternObject = {|
```

Comment:
I don't like adding "Object" because everything in JS is an object, hence why it was called `PathPattern`. How about `TestSelectionConfig` or `TestSelectionPattern` because that is what it actually is?

Note: Previous names of this were `PathPatternInfo` and `PathInfo` or things like it. It's a tough one.